rust/src/main.rs

```diff
@@ -76,37 +109,147 @@ fn setup_logging() -> Result<(), fern::InitError> {
         }).level(level_filter)
         .chain(io::stderr());
 
-    let path_result = path_for_log_file(XI_LOG_FILE);
-    // If the logging_file_path returned successfully, add the logfile capability to fern
-    if let Ok(logging_file_path) = &path_result {
+    if let Some(logging_file_path) = &logging_path {
+        // Try to create the parent directories for the logging file in the logging file path
```

Comment:
ditto my earlier comment, also for the next comment on 116; in both of these cases I think the code speaks for itself.

(I think this is true of a few other places as well, but it's all subjective, I'm not going to try and point them all out.)

Reply:
See other comment for 116